lilypad/_opentelemetry/_opentelemetry_groq/patch.py

```diff
@@ -0,0 +1,269 @@
+        ) as span:
+            if span.is_recording():
+                for message in kwargs.get("messages", []):
+                    set_message_event(span, message)
+            try:
+                result = await wrapped(*args, **kwargs)
+
+                if kwargs.get("stream", False):
+                    # Convert list to iterator if necessary
+                    if isinstance(result, list):
+                        result = iter(result)
+
+                    class WrappedAsyncStream(AsyncStreamProtocol):
+                        """A wrapper for async streams that ensures proper protocol implementation."""
+
+                        def __init__(self) -> None:
+                            """Initialize the async stream wrapper."""
+                            self._iterator: AsyncIterator[Any] | Iterator[Any] = result
+
+                        def __aiter__(self) -> AsyncIterator[Any]:
+                            """Return self as an async iterator."""
+                            return self
+
+                        async def __anext__(self) -> Any:
+                            """Get the next item from the stream."""
+                            try:
+                                if isinstance(self._iterator, AsyncIterator):
+                                    return await self._iterator.__anext__()
+                                if isinstance(self._iterator, Iterator):
+                                    try:
+                                        return next(self._iterator)
+                                    except StopIteration:
+                                        raise StopAsyncIteration
+                                # Handle async generators
+                                if hasattr(self._iterator, "asend"):
+                                    return await self._iterator.asend(None)  # type: ignore
+                                raise StopAsyncIteration
+                            except (StopIteration, StopAsyncIteration):
+                                raise StopAsyncIteration
+
+                        async def aclose(self) -> None:
+                            """Close the async stream."""
+                            if hasattr(self._iterator, "aclose"):
+                                await self._iterator.aclose()  # type: ignore
+
+                    # Ensure stream implements required protocol
+                    if not hasattr(result, "__aiter__"):
+                        result = WrappedAsyncStream()
+                    elif not isinstance(result, WrappedAsyncStream):
+                        # Wrap existing async iterators to ensure consistent behavior
+                        wrapped_stream = WrappedAsyncStream()
+                        wrapped_stream._iterator = result
+                        result = wrapped_stream
+
+                    return AsyncStreamWrapper(
+                        span=span,
+                        stream=cast(AsyncStreamProtocol, result),
```

Comment:
Why are you using `AsyncStreamProtocol` instead of the common one used in utils. You should be able to set `stream=result` and use the common StreamWrapper